Smith.MatrixSdk.Example/Application.cs

```diff
@@ -13,24 +14,32 @@ internal class Application : ConsoleAppBase
     {
         private readonly ILogger _logger;
         private readonly HttpClient _httpClient;
+        private readonly CancellationTokenSource _ctrl_c_cts;
 
         public Application(ILogger<Application> logger, HttpClient httpClient)
         {
             _logger = logger;
             _httpClient = httpClient;
+            _ctrl_c_cts = new CancellationTokenSource();
+
+            Console.CancelKeyPress += delegate {
+                _ctrl_c_cts.Cancel();
+            };
         }
 
         [UsedImplicitly]
         public async Task Run(string user, string password, string homeserver = "https://matrix.org")
         {
             var client = new MatrixClient(_logger, _httpClient, new Uri(homeserver));
-            var loginResponse = await client.Login(user, password);
+            var loginResponse = await client.Login(user, password, _ctrl_c_cts.Token);
             Console.WriteLine(loginResponse);
 
             await client.StartEventPolling(loginResponse.AccessToken.NotNull(), TimeSpan.FromSeconds(5))
                 .Do(syncResponse =>
                 {
                     Console.WriteLine($"Next batch: {syncResponse.NextBatch}");
+
+                    _ctrl_c_cts.Token.ThrowIfCancellationRequested();
```

Comment:
What's wrong here is that the application wouldn't terminate if there's an in-flight request. See: in your variant, it only checks the request _after_ receiving a response, any response. And if the response is waiting on a timeout, then after pressing `Ctrl+C`, it won't be terminated. I believe that my implementation in #14 will handle that better (provided my understanding of the cancellation token flow in the Rx framework is correct).